### PR TITLE
Implement missing API & refactor for View

### PR DESCRIPTION
**Pull request by infosia**
Fix for [TIMOB-19921](https://jira.appcelerator.org/browse/TIMOB-19921) and [TIMOB-19948](https://jira.appcelerator.org/browse/TIMOB-19948)

- Implement missing View properties & events
- Refactoring View events
- Add View unit tests
- Use Font cache for TextArea and Button

## Newly supported View properties

- backgroundFocusedColor
- backgroundFocusedImage
- backgroundGradient
- backgroundSelectedColor
- backgroundSelectedImage
- borderColor
- borderWidth

```javascript
var win = Ti.UI.createWindow();
var view = Ti.UI.createView({width:'50%', height:'50%'});

//view.backgroundImage = 'http://www.appcelerator.com/wp-content/themes/appc-rwd/assets/media/images/logo.png';
//view.backgroundColor = 'green';
//view.backgroundFocusedColor = "green";
//view.backgroundSelectedColor = "blue";
//view.backgroundSelectedImage = "Square71x71Logo.png";
//view.backgroundFocusedImage = "Square71x71Logo.png";

view.backgroundGradient = {
    type: 'linear',
    startPoint: { x: '0%', y: '50%' },
    endPoint: { x: '100%', y: '100%' },
    colors: [{ color: 'red', offset: 0.0 }, { color: 'blue', offset: 0.25 }, { color: 'red', offset: 1.0 }],
}
win.add(view);
win.open();
```

## Known issue

Some properties & events doesn't work because certain events aren't fired on some Windows components.

### background color
component | focused background | selected background |  normal background
------------ | ------------ | ------------- | -------------
Ti.UI.Window | :x: | :x: | :o: | 
Ti.UI.View | :x: | :x: | :o: | 
Ti.UI.Label | :x: | :x: | :x:
Ti.UI.Button | :o: | :o: | :o:
Ti.UI.Slider | :o: | :o: | :o:
Ti.UI.Switch | :o: | :o: | :o:
Ti.UI.TabGroup | :o: | :o: | :o:
Ti.UI.TextField | :o: | :x: | :o:
Ti.UI.TextArea | :o: | :x: | :o:
Ti.UI.ListView | :x: | :o: | :o: | 
Ti.UI.TableView | :x: | :o: | :o: | 
Ti.UI.Picker | :x: | :o: | :o: | 
Ti.UI.ProgressBar | :x: | :o: | :o: | 
Ti.UI.ScrollableView | :x: | :o: | :o: | 
Ti.UI.ScrollView | :x: | :o: | :o: | 

## events

component | focused/blur | touch event
------------ | ------------ | ------------- 
Ti.UI.Window | :o: | :o:
Ti.UI.View | :x: | :o:
Ti.UI.Label | :x: | :o:
Ti.UI.Button | :o: | :o:
Ti.UI.Slider | :o: | :o: 
Ti.UI.Switch | :o: | :o:
Ti.UI.TabGroup | :o: | :o:
Ti.UI.TextField | :o: | :x:
Ti.UI.TextArea | :o: | :x:
Ti.UI.ListView | :x: | :o:
Ti.UI.TableView | :x: | :o:
Ti.UI.Picker | :x: | :o:
Ti.UI.ProgressBar | :x: | :o:
Ti.UI.ScrollableView | :x: | :o:
Ti.UI.ScrollView | :x: | :o:

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
test this please.

**Comment by sgtcoolguy** [comment]
test this please.

File: Source/TitaniumKit/src/UI/View.cpp [hunk 1]
```diff
@@ -70,12 +70,12 @@ namespace Titanium
 
 		void View::blur() 
 		{
-			fireEvent("blur");
+			layoutDelegate__->blur();
```
Comment (sgtcoolguy) [inline Source/TitaniumKit/src/UI/View.cpp]: I don't think the LayoutDelegate fires the events for blur and focus, in it's blur/focus methods. Does it? I assume we still need to fire these...

File: Source/TitaniumKit/src/UI/Point.cpp [hunk 2]
```diff
@@ -17,10 +18,34 @@ namespace Titanium
 		{
 			Point point;
 			if (object.HasProperty("x")) {
-				point.x = static_cast<double>(object.GetProperty("x"));
+				const auto xobj = object.GetProperty("x");
+				if (xobj.IsString()) {
```
Comment (sgtcoolguy) [inline Source/TitaniumKit/src/UI/Point.cpp]: refactor out the common code for handling x and y values?

File: Source/TitaniumKit/src/UI/View.cpp (hunk 1 above)
Comment (infosia) [inline Source/TitaniumKit/src/UI/View.cpp]: These are kind of special methods that is used internally (mainly from  `Window`). Yes, Titanium Views are still responsible for firing `blur/focus` Titanium events (typically through `GotFocus/LostFocus` Xaml events).

File: Source/TitaniumKit/src/UI/Point.cpp (hunk 2 above)
Comment (infosia) [inline Source/TitaniumKit/src/UI/Point.cpp]: Can this be simpler? Maybe this is little bit confusing that this updates `x_percent` (string) *or* `x` (double).  When`x_percent` exists `x` should be updated later by Point owner.

**Pull request merged**